tests/rbd_mirror/test_rbd_mirror_connection_metrics.py

```diff
@@ -0,0 +1,149 @@
+        ceph_AsyncMessenger_Worker_msgr_connection_idle_timeouts{id="0"} 128
+
+        # TYPE ceph_AsyncMessenger_Worker_msgr_connection_ready_timeouts counter
+        ceph_AsyncMessenger_Worker_msgr_connection_ready_timeouts{id="0"} 0
+
+    Returns:
+        0 - if test case pass
+        1 - if test case fails
+    """
+    # connection metrics to validate
+    connection_metrics = [
+        "ceph_AsyncMessenger_Worker_msgr_connection_idle_timeouts",
+        "ceph_AsyncMessenger_Worker_msgr_connection_ready_timeouts",
+    ]
+
+    node_ip = admin_node.ip_address
+    cmd = f"curl http://{node_ip}:{port}/metrics | grep ceph_AsyncMessenger_Worker_msgr"
+
+    out = admin_node.exec_command(cmd=cmd, output=True, sudo=True)
+    lines = out[0].split("\n")
+
+    # Dictionary to store parsed metrics
+    metrics = {}
+
+    # Iterate over lines and extract metrics
+    for line in lines:
+        if line.startswith("#") or not line.strip():
+            # Skip comments and empty lines
+            continue
+
+        # Split the line into parts using whitespace
+        parts = line.split()
+
+        # Extract relevant information
+        metric_name = parts[0]
+        metric_value = parts[1]
+
+        # Add to the metrics dictionary
+        metrics[metric_name] = int(metric_value)
+
+    # Check if metrics are available in the response
+    if not metrics:
+        log.error("Error: No metrics found in the response.")
+        return 1
+
+    # Validate each connection metric
+    for counter in connection_metrics:
+        if not any(counter in metric_name for metric_name in metrics):
+            log.error(f"Error: Metric {counter} not found in the response {metrics}")
+            return 1
+
+    log.info(f"Network connection Metrics avialble are {metrics}")
+    return 0
+
+
+def run(**kw):
+    """Verification of Performance counter for rbd mirror network
+    connection health metrics.
+
+    Args:
+        **kw: test data
+
+    Returns:
+        0 - if test case pass
+        1 - if test case fails
+    """
+
+    try:
+        log.info(
+            "Starting test CEPH-83575566 - Performance counter for rbd mirror\
+             network connection health metrics."
+        )
+        config = kw.get("config")
+
+        mirror1 = rbdmirror.RbdMirror(
+            kw.get("ceph_cluster_dict").get("ceph-rbd1"), config
+        )
+        mirror2 = rbdmirror.RbdMirror(
+            kw.get("ceph_cluster_dict").get("ceph-rbd2"), config
+        )
+
+        poolname = mirror1.random_string() + "_test_pool"
+        imagename = mirror1.random_string() + "_test_image"
+        metrics_port = config.get("ceph_exporter_port", "9926")
+
+        mirror1.initial_mirror_config(
+            mirror2,
+            poolname=poolname,
+            imagename=imagename,
+            imagesize=config.get("imagesize", "10G"),
+            io_total=config.get("io_total", "4G"),
+            mode=config.get("mode", "journal"),
```

Comment:
for mode, default value should be pool and for mirrormode default value could either be empty or journal

Reply:
sorry updated.